cmd/pebble/cmd_enter.go

```diff
@@ -0,0 +1,186 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/canonical/pebble/internal/logger"
+	"github.com/jessevdk/go-flags"
+)
+
+const shortEnterHelp = "Run pebble command in new pebble environment"
+const longEnterHelp = `
+The enter command behaves like the run command with the possibility of running
+a single pebble command after starting the service manager. It's intended to be
+used as entrypoint in container images.
+
+When no subcommand is given, the behavior is identical to the run command.
+
+When a subcommand is given, the service manager is started in the background
+without starting "autostart" services (i.e. run --hold, unless --run flag is
+given), and the subcommand is run as a pebble client command that can
+communicate with the service manager instance in the background. The service
+manager is shut down after the commands finishes unless noted otherwise below.
```

Comment:
This is doing a reasonable job of describing the behavior, but as Ben already spotted it is also very confusing because the actual behavior of any one command is hard to grasp because it's an ANDing of a number of different conditional paragraphs that may affect the given command or not.  The agreement we've reached at the end of our several meetings was supposed to be one where the behavior of enter would be the most natural option for the given command at hand, so if we did that justice, we shoud be able to have a more straightforward description here too.

With that said, let's leave that aside for now and focus on the implementation. Once this is ready to go in, we can do a pass together and find the best way to word this out.

Reply:
I agree it's not exactly straightforward read.